src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -89,25 +116,36 @@ private static void openEditor(Player p, Block projector) {
             return false;
         });
 
-        menu.addItem(1, new CustomItem(Material.CLOCK, "&7Offset: &e" + DoubleHandler.fixDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + 1.0D), "", "&rLeft Click: &7+0.1", "&rRight Click: &7-0.1"));
-        menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            double offset = DoubleHandler.fixDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + (action.isRightClicked() ? -0.1F : 0.1F));
+        menu.addItem(9, new CustomItem(Material.CLOCK, "&cReset Position", "", "&eClick &7to reset hologram location"));
+        menu.addMenuClickHandler(9, (pl, slot, item, action) -> {
             ArmorStand hologram = getArmorStand(projector, true);
-            Location l = new Location(projector.getWorld(), projector.getX() + 0.5, projector.getY() + offset, projector.getZ() + 0.5);
-            hologram.teleport(l);
+            hologram.teleport(projector.getLocation().add(0.5, 0.5, 0.5));
 
-            BlockStorage.addBlockInfo(projector, OFFSET_PARAMETER, String.valueOf(offset));
+            BlockStorage.addBlockInfo(projector, X_OFFSET_PARAMETER, "0.5");
+            BlockStorage.addBlockInfo(projector, Y_OFFSET_PARAMETER, "0.5");
+            BlockStorage.addBlockInfo(projector, Z_OFFSET_PARAMETER, "0.5");
             openEditor(pl, projector);
             return false;
         });
 
+        createSelector(2, HeadTexture.RED_UP_ARROW.getAsItemStack(), 'X', '+', X_OFFSET_PARAMETER, 0.1F, 0.5F, projector, menu);
+        createSelector(11, HeadTexture.RED_DOWN_ARROW.getAsItemStack(), 'X', '-', X_OFFSET_PARAMETER, -0.1F, -0.5F, projector, menu);
+
+        createSelector(4, HeadTexture.LIME_UP_ARROW.getAsItemStack(), 'Y', '+', Y_OFFSET_PARAMETER, 0.1F, 0.5F, projector, menu);
+        createSelector(13, HeadTexture.LIME_DOWN_ARROW.getAsItemStack(), 'Y', '-', Y_OFFSET_PARAMETER, -0.1F, -0.5F, projector, menu);
+
+        createSelector(6, HeadTexture.BLUE_UP_ARROW.getAsItemStack(), 'Z', '+', Z_OFFSET_PARAMETER, 0.1F, 0.5F, projector, menu);
+        createSelector(15, HeadTexture.BLUE_DOWN_ARROW.getAsItemStack(), 'Z', '-', Z_OFFSET_PARAMETER, -0.1F, -0.5F, projector, menu);
```

Comment:
Not the biggest fan of using characters here...
I think you should create an enum for this, positive x, negative x, positive y, etc...
This could be placed into the `utils.holograms` package

Reply:
What if I grabbed the sign off of the double

Reply:
I mean it would be better. But an enum would still be ideal for this. Especially because passing the axis via a character is not really a good design choice.